File: solutions/cpp/isbn-verifier/2/isbn_verifier.cpp
#include "isbn_verifier.h"

#include <vector>

using namespace std;

namespace isbn_verifier {

    bool is_valid(string isbn) {
        unsigned int factor=10;
        int sum = 0;
        for(auto ch : isbn) {
            if (ch=='-') 
                continue;
            bool valid = isdigit(ch) ||
                        (ch=='X' && factor==1);
            if(!valid) 
                return false;
            else {
                int val = ch == 'X' ? 10 : ch - '0'; 
                sum += val*factor; } 
            factor--;
        }
        return factor==0 && 0==sum%11;
    }


}  // namespace isbn_verifier

File: solutions/python/isbn-verifier/2/isbn_verifier.py
def is_valid(isbn):
    factor = 10
    sum = 0
    for ch in isbn:
        if ch == '-':
            continue        
        if ch.isdigit():
            sum += factor*int(ch)
        elif ch=='X' and factor==1: 
            sum += 10
        else:
            return False
        factor -= 1                 
    return factor==0 and sum % 11 == 0
        
        
